src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -274,7 +274,7 @@ foam.CLASS({
         Object id = obj.getProperty("id");
         if ( id != null &&
              ! SafetyUtil.isEmpty(id.toString()) ) {
-          FObject old = dao.inX(getX()).find(id);
+          FObject old = dao.inX(x).find(id);
```

Comment:
By reading the comment on line 273, seems like using getX() was intentional.

Reply:
hmm, think we'll need to add an auth check to explicitly check for the read permission here.